databricks/koalas/base.py

```diff
@@ -391,7 +399,11 @@ def hasnans(self):
         sdf = self._internal.spark_frame
         scol = self.spark.column
 
-        return sdf.select(F.max(scol.isNull() | F.isnan(scol))).collect()[0][0]
+        if isinstance(self.spark.data_type, BooleanType):
```

Comment:
What if the other data types, such as `StringType`, `DateType`, or `TimestampType`?

Reply:
Ah, I see. `np.nan` is available only for the `DoubleType` column. Thanks! :)